packages/e2e-tests/specs/block-grouping.test.js

```diff
@@ -0,0 +1,234 @@
+	await page.keyboard.type( 'Second Paragraph' );
+
+	await insertBlock( 'Paragraph' );
+	await page.keyboard.type( 'Third Paragraph' );
+}
+
+async function insertBlocksOfMultipleTypes() {
+	await insertBlock( 'Heading' );
+	await page.keyboard.type( 'Group Heading' );
+
+	await insertBlock( 'Image' );
+
+	await insertBlock( 'Paragraph' );
+	await page.keyboard.type( 'Some paragraph' );
+}
+
+describe( 'Block Grouping', () => {
+	beforeAll( async () => {
+		await createNewPost();
+	} );
+
+	beforeEach( async () => {
+		// Remove all blocks from the post so that we're working with a clean slate
+		await page.evaluate( () => {
+			const blocks = wp.data.select( 'core/editor' ).getBlocks();
+			const clientIds = blocks.map( ( block ) => block.clientId );
+			wp.data.dispatch( 'core/editor' ).removeBlocks( clientIds );
+		} );
+	} );
+
+	describe( 'Group creation', () => {
+		it( 'creates a group from multiple blocks of the same type via block transforms', async () => {
+			// Creating test blocks
+			await insertBlocksOfSameType();
+
+			// Multiselect via keyboard.
+			await pressKeyWithModifier( 'primary', 'a' );
+			await pressKeyWithModifier( 'primary', 'a' );
+
+			await transformBlockTo( 'Group' );
+
+			expect( await getEditedPostContent() ).toMatchSnapshot();
+		} );
+
+		it( 'creates a group from multiple blocks of different types via block transforms', async () => {
+			// Creating test blocks
+			await insertBlocksOfMultipleTypes();
+
+			// Multiselect via keyboard.
+			await pressKeyWithModifier( 'primary', 'a' );
+			await pressKeyWithModifier( 'primary', 'a' );
+
+			await transformBlockTo( 'Group' );
+
+			expect( await getEditedPostContent() ).toMatchSnapshot();
+		} );
+
+		it( 'creates a group from multiple blocks of the same type via options toolbar', async () => {
+			// Creating test blocks
+			await insertBlocksOfSameType();
+
+			// Multiselect via keyboard.
+			await pressKeyWithModifier( 'primary', 'a' );
+			await pressKeyWithModifier( 'primary', 'a' );
+
+			await clickBlockToolbarButton( 'More options' );
+
+			const groupButton = await page.waitForXPath( '//button[text()="Group"]' );
+			await groupButton.click();
+
+			expect( await getEditedPostContent() ).toMatchSnapshot();
+		} );
+
+		it( 'creates a group from multiple blocks of different types via options toolbar', async () => {
+			// Creating test blocks
+			await insertBlocksOfMultipleTypes();
+
+			// Multiselect via keyboard.
+			await pressKeyWithModifier( 'primary', 'a' );
+			await pressKeyWithModifier( 'primary', 'a' );
+
+			await clickBlockToolbarButton( 'More options' );
+
+			const groupButton = await page.waitForXPath( '//button[text()="Group"]' );
+			await groupButton.click();
+
+			expect( await getEditedPostContent() ).toMatchSnapshot();
+		} );
+	} );
+
+	describe( 'Ungrouping', () => {
+		it( 'ungroups an existing group via options menu', async () => {
+			// Create a Group
+			await insertBlocksOfMultipleTypes();
```

Comment:
This could be combined into one test along with `creates a group from multiple blocks of different types via options toolbar` as they mostly perform the same steps. There's no restriction to the amount of snapshots in a single test case.

The amount of time it takes to run CI tests is the main concern here, anything that can be done to reduce duplicated steps is a positive.

Reply:
Ok thanks for advice. Fixed.